### ISSUE TITLE
Allow CLI configuration to be saved to a file

### ISSUE THREAD

**Issue by amisevsk** [jozu-ai/kitops#419]
**Describe the problem you're trying to solve**
As the CLI becomes more complex and gains more options, it would be convenient to be able to set some/all of them persistently in a configuration file. This will become increasingly useful as e.g. more networking options are added.

**Describe the solution you'd like**
Kit should provide a `kit config` command to allow configuring common options available as flags and saving that configuration between invocations. This configuration should be saved to some file (maybe in the `$KITOPS_HOME` directory, or separate Kit config directory) and these settings should be used instead of default values for corresponding flags.

**Describe alternatives you've considered**
Mainly, deferring this work until there are enough flags we're specifying on every invocation where the moderately increased complexity of supporting a config file makes sense. 

**Additional context**
Related: #365 -- we want to be able to persistently disable update notifications, for example.

**Comment by SkySingh04** [comment]
This sounds like a fun and interesting issue to work on! This will require a little bit of work but lets do this!

**Comment by bartoszmajsak** [comment]
This is my first activity here, so 👋🏻 :)

@amisevsk @gorkem One thing I wonder - was [`viper` integration with Cobra](https://github.com/spf13/viper#working-with-flags) considered? That allows to define config sources (and their precedence) like env vars, and files (multiple formats) seamlessly.

**Comment by gorkem** [comment]
@bartoszmajsak we used viper on the early days for a small case.  @amisevsk found a way to do it without it and we removed. Perhaps, we do have more of a use for it now to reconsider.

**Comment by amisevsk** [comment]
Yeah, we considered viper but I found (at least early on) that it was not a great fit. It may make sense to use now, though we've still only got ~2-3 config settings (log level, progress bars), that make sense to set persistently. I agree it might be worth revisiting (though I remember thinking actually using viper in commands was somewhat cumbersome)

**Comment by bupd** [comment]
> I remember thinking actually using viper in commands was somewhat cumbersome

@amisevsk 
A better approach might be to move the viper setup into a separate package and have it run on every command execution  
(ie,  global config for the cli). That way, we don’t need to configure it within each CLI command.

**Comment by amisevsk** [comment]
IIRC grouping viper setup in one package was really annoying, since each subcommand in Kit has its own args, and getting that to work with Viper required the same sort of machinery [pulling from context does](https://github.com/kitops-ml/kitops/blob/main/pkg/cmd/pull/cmd.go#L48-L52). It has been a while since I looked into it, so it's maybe worth a second glance though.

Basically, it felt like a fairly large and complicated dependency for our use case, which _should_ be relatively simple.

**Comment by omarbassem128** [comment]
I am interested in claiming this issue and I am currently digging into the codebase to deeply understand it.

**Comment by omarbassem128** [comment]
I have finished the config command's implementation and integration with the root options. I've also tested it manually in the terminal and it is  working as expected. Are automated tests necessary or can I open a pull request without them ? @amisevsk

**Comment by amisevsk** [comment]
Tests are always good to have, but aren't strictly necessary for an initial PR. Feel free to open a PR and we can discuss there 👍
